File: app/src/main/java/link/couple/jin/couplelink/data/UserClass.java
package link.couple.jin.couplelink.data;

import android.support.annotation.NonNull;

import java.util.HashMap;
import java.util.Map;

/**
 * 회원가입시 데이터베이스에 이메일과 이름 저장
 */

public class UserClass {

    public String email;
    public String username;
    public String couple = "";
    public boolean isCouple = false;
    public boolean isCoupleConnect = false;
    public String uid;

    public UserClass(){}

    public UserClass(@NonNull String email, @NonNull String username, @NonNull String couple, @NonNull boolean isCouple,@NonNull boolean isCoupleConnect){
        this.email = email;
        this.username = username;
        this.couple = couple;
        this.isCouple = isCouple;
        this.isCoupleConnect = isCoupleConnect;
    }

    public Map<String, Object> toMap() {
        HashMap<String, Object> result = new HashMap<>();
        result.put("email", email);
        result.put("username", username);
        result.put("couple", couple);
        result.put("isCouple", isCouple);
        result.put("isCoupleConnect", isCoupleConnect);
        return result;
    }

}

File: app/src/main/java/link/couple/jin/couplelink/data/UserClass.kt
package link.couple.jin.couplelink.data

import java.util.HashMap

/**
 * 회원가입시 데이터베이스에 이메일과 이름 저장
 */

class UserClass {

    var email: String
    var username: String
    var couple = ""
    var isCouple = false
    var isCoupleConnect = false
    var uid: String? = null
    var fcm = ""

    constructor() {}

    constructor(email: String, username: String, couple: String, isCouple: Boolean, isCoupleConnect: Boolean) {
        this.email = email
        this.username = username
        this.couple = couple
        this.isCouple = isCouple
        this.isCoupleConnect = isCoupleConnect
    }

    fun toMap(): Map<String, Any> {
        val result = HashMap<String, Any>()
        result["email"] = email
        result["username"] = username
        result["couple"] = couple
        result["isCouple"] = isCouple
        result["isCoupleConnect"] = isCoupleConnect
        result["fcm"] = fcm
        return result
    }

}
